Version 1.12.0 breaks azuredevops_team  .. it wil never reach "Synced" state.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

Terraform 1.12.0
Provider 1.12.0

### Affected Resource(s)

azuredevops_team

### Debug Output

```
2025-12-11T13:19:18.9686320Z 2025-12-11T13:19:18.967Z [TRACE] dag/walk: vertex "module.generic_devops_projects (close)" is waiting for "module.generic_devops_projects.module.teams (close)"
2025-12-11T13:19:18.9755267Z 2025-12-11T13:19:18.975Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:18 [TRACE] Waiting 10s before next try: timestamp=2025-12-11T13:19:18.975Z
2025-12-11T13:19:18.9806526Z 2025-12-11T13:19:18.980Z [TRACE] dag/walk: vertex "root" is waiting for "module.generic_devops_projects[\"XXXXX\"].module.teams[\"XXXX\"].output.id"
2025-12-11T13:19:18.9941955Z 2025-12-11T13:19:18.993Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:18 [TRACE] Waiting 10s before next try: timestamp=2025-12-11T13:19:18.993Z
2025-12-11T13:19:18.9943292Z 2025-12-11T13:19:18.994Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:18 [TRACE] Waiting 10s before next try: timestamp=2025-12-11T13:19:18.994Z
2025-12-11T13:19:19.0703317Z 2025-12-11T13:19:19.069Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:19 [TRACE] Waiting 10s before next try: timestamp=2025-12-11T13:19:19.069Z
2025-12-11T13:19:19.3029489Z 2025-12-11T13:19:19.302Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:19 [WARN] WaitForState timeout after 30m0s: timestamp=2025-12-11T13:19:19.302Z
2025-12-11T13:19:19.3030508Z 2025-12-11T13:19:19.302Z [INFO]  provider.terraform-provider-azuredevops_v1.12.0: 2025/12/11 13:19:19 [WARN] WaitForState starting 30s refresh grace period: timestamp=2025-12-11T13:19:19.302Z
2025-12-11T13:19:19.3031687Z 2025-12-11T13:19:19.302Z [TRACE] provider.terraform-provider-azuredevops_v1.12.0: Called downstream: tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:980 @module=sdk.helper_schema tf_provider_addr=registry.terraform.io/microsoft/azuredevops tf_resource_type=azuredevops_team tf_req_id=ecc66fda-ce58-2b88-9b61-440f14996233 timestamp=2025-12-11T13:19:19.302Z
2025-12-11T13:19:19.3033179Z 2025-12-11T13:19:19.302Z [TRACE] provider.terraform-provider-azuredevops_v1.12.0: Received downstream response: @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:42 @module=sdk.proto diagnostic_warning_count=0 tf_provider_addr=registry.terraform.io/microsoft/azuredevops tf_req_duration_ms="1.800541e+06" tf_resource_type=azuredevops_team tf_rpc=ApplyResourceChange diagnostic_error_count=1 tf_proto_version=5.8 tf_req_id=ecc66fda-ce58-2b88-9b61-440f14996233 timestamp=2025-12-11T13:19:19.302Z
2025-12-11T13:19:19.3034533Z 2025-12-11T13:19:19.302Z [ERROR] provider.terraform-provider-azuredevops_v1.12.0: Response contains error diagnostic: tf_resource_type=azuredevops_team diagnostic_severity=ERROR diagnostic_summary="waiting for state change for team 67efb6dd-66b9-4d14-b2d6-c5f9a4d744c6 in project 8c391b0a-72b7-41d7-b6b2-d57ec53f85bc. timeout while waiting for state to become 'Synched' (last state: 'Waiting', timeout: 30m0s) " tf_provider_addr=registry.terraform.io/microsoft/azuredevops tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_detail="" tf_proto_version=5.8 tf_req_id=ecc66fda-ce58-2b88-9b61-440f14996233 timestamp=2025-12-11T13:19:19.302Z
2025-12-11T13:19:19.3035929Z 2025-12-11T13:19:19.302Z [TRACE] provider.terraform-provider-azuredevops_v1.12.0: Served request: tf_proto_version=5.8 tf_provider_addr=registry.terraform.io/microsoft/azuredevops tf_req_id=ecc66fda-ce58-2b88-9b61-440f14996233 tf_resource_type=azuredevops_team tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:879 @module=sdk.proto timestamp=2025-12-11T13:19:19.
```

### Expected Behavior

The Script to complete.

### Actual Behavior

The script suddenly stops processing azuredevops_team  resources, and just keeps looping forever "still modifying..." until the timeout of 30 minutes is expired. 

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

It works perfectly in version 1.11.2, in v 1.12.0 it breaks.
After reverting back to 1.11.2 everything worked as normal immediately.

The script uses large amounts of users / teams / groups.

### References

--- a/azuredevops/internal/service/permissions/utils/namespaces.go
+++ b/azuredevops/internal/service/permissions/utils/namespaces.go
@@ -240,19 +240,17 @@ func (sn *SecurityNamespace) GetActionDefinitions() (*map[string]security.Action
 }
 
 func (sn *SecurityNamespace) GetAccessControlList(descriptorList *[]string) (*security.AccessControlList, error) {
-	if descriptorList == nil || len(*descriptorList) == 0 {
-		return nil, nil
+	var descriptors *string = nil
+	if descriptorList != nil && len(*descriptorList) > 0 {
+		val := linq.From(*descriptorList).
+			Aggregate(func(r interface{}, i interface{}) interface{} {
+				if r.(string) == "" {
+					return i
+				}
+				return r.(string) + "," + i.(string)
+			}).(string)
+		descriptors = &val
 	}
-
-	var descriptors *string
-	val := linq.From(*descriptorList).
-		Aggregate(func(r interface{}, i interface{}) interface{} {
-			if r.(string) == "" {
-				return i
-			}
-			return r.(string) + "," + i.(string)
-		}).(string)
-	descriptors = &val
 
 	bTrue := true
 	acl, err := sn.securityClient.QueryAccessControlLists(sn.context, security.QueryAccessControlListsArgs{

